File: 535-encode-and-decode-tinyurl/535-encode-and-decode-tinyurl.java
public class Codec {
    List<String> l;
    public Codec(){
        l = new ArrayList<>();
    }
    // Encodes a URL to a shortened URL.
    public String encode(String longUrl) {
        l.add(longUrl);
        return String.valueOf(l.size()-1);
    }

    // Decodes a shortened URL to its original URL.
    public String decode(String shortUrl) {
        int idx = Integer.parseInt(shortUrl);
        return l.get(idx);
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(url));

File: 535-encode-and-decode-tinyurl/535-encode-and-decode-tinyurl.py
class Codec:
    def __init__(self):
        self.myMap = {}
        self.count = 0
    def encode(self, longUrl: str) -> str:
        """Encodes a URL to a shortened URL.
        """
        self.myMap[self.count] = longUrl
        self.count+=1
        return str(self.count-1)
        
    def decode(self, shortUrl: str) -> str:
        """Decodes a shortened URL to its original URL.
        """
        return self.myMap[int(shortUrl)]
        

# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(url))